src/main/java/org/fogbowcloud/manager/core/rest/services/ComputeOrdersService.java

```diff
@@ -0,0 +1,75 @@
+package org.fogbowcloud.manager.core.rest.services;
+
+import org.fogbowcloud.manager.core.controllers.ApplicationController;
+import org.fogbowcloud.manager.core.datastructures.SharedOrderHolders;
+import org.fogbowcloud.manager.core.exceptions.ComputeOrdersServiceException;
+import org.fogbowcloud.manager.core.models.orders.ComputeOrder;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.token.Token;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+@Service
+public class ComputeOrdersService {
```

Comment:
This class seems to me that can be used by the others orders types, you can refactor it in a generical way.